packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2192,35 +2247,56 @@ class AnnotatedRegionLayer<T> extends ContainerLayer {
   /// If not provided, offset defaults to [Offset.zero].
   ///
   /// Ignored if [size] is not set.
+  ///
+  /// Unlike [OffsetLayer], this parameter [offset] does not affect the layer's
+  /// children.
   final Offset offset;
 
-  @override
-  S find<S>(Offset regionOffset) {
-    final S result = super.find<S>(regionOffset);
-    if (result != null)
-      return result;
-    if (size != null && !(offset & size).contains(regionOffset))
-      return null;
-    if (T == S) {
-      final Object untypedResult = value;
-      final S typedResult = untypedResult;
-      return typedResult;
-    }
-    return null;
-  }
+  /// Whether this layer should prevent layers visually behind it from being
+  /// hit during a hit test.
+  ///
+  /// If [opaque] is true, this layer will absorb the hit if this layer contains
+  /// the target position and has the requested annotation.
+  ///
+  /// If [opaque] is false, this layer will not change the absorption value
+  /// returned by its children.
+  ///
+  /// This defaults to false.
+  ///
+  /// See also:
+  ///
+  ///  * [HitTestBehavior], which controls similar logic when hit-testing in the
+  ///    render tree.
+  final bool opaque;
 
+  /// Adds annotations from the children, then from this layer.
+  ///
+  /// The children's annotations are always added.
+  ///
+  /// The layer's annotation [value] is added if the type `S` matches `T`, and
+  /// the target position meets the requirement defined by [size] and [offset].
+  ///
+  /// The return value is true if any child returns true, or the layer's
+  /// annotation is added and [opaque] is true.
   @override
-  Iterable<S> findAll<S>(Offset regionOffset) {
-    final Iterable<S> childResults = super.findAll<S>(regionOffset);
-    if (size != null && !(offset & size).contains(regionOffset)) {
-      return childResults;
+  @protected
+  bool findAnnotations<S>(AnnotationResult<S> result, Offset localPosition, { @required bool onlyFirst }) {
+    bool isAbsorbed = super.findAnnotations(result, localPosition, onlyFirst: onlyFirst);
```

Comment:
What's the relationship between `onlyFirst` and `opaque`? It seems like you should specify true for `onlyFirst` if `opaque` is true, to avoid searching further than you need to.  But maybe I'm just not understanding what the relationship is...

Reply:
Parameter `onlyFirst` is a property of the tree walking _process_(function call), while `opaque` is a property of _layers_.

If `onlyFirst` is true, then the layers' `opaque` doesn't actually matter.

If `onlyFirst` is not true, then `opaque` can change results. Consider the following case, where C is parent to A and B, and we call `C.findAnnotations` with the position of the pointer. 
```
       | Pointer
       v 
      ------- B
           |
A ------   |
   |       |
C------------
```
If B is opaque, then the result is `[B, C]`; if B is not opaque, then the result is `[B, A, C]`. (Apparently, if `onlyFirst` is true, then the result is always `[B]`.)

Reply:
I'll appreciate it if you can help me with the documentation to make it more comprehensible :)

Reply:
Sure, no problem!

So, why isn't the result just `[B]` if B is opaque? I thought that was the point of it being opaque, that it stopped on the first hit.

Reply:
Opacity only controls the relationship between siblings. (Parents usually returns if any of their children returns true.)

In other words, if one layer is hit, then its ancestors are _always_ hit too.

(This algorithm is similar to `HitTestBehavior.opaque` in render object hit testing.)